docs/user-guide/troubleshooting/credentials.md

```diff
@@ -0,0 +1,76 @@
+# Troubleshooting credentials issues
+
+!!! info "Make sure you read [general troubleshooting](../general) page before trying out anything else."
+
+## Are you using IAM or SSO?
+Leverage supports two methods for getting AWS credentials: IAM and SSO. We are progressively favoring SSO over IAM, only using the latter as a fallback option.
+
+SSO is enabled through the [common.tfvars](https://github.com/binbashar/le-tf-infra-aws/blob/master/config/common.tfvars.example) file on this line:
+```
+sso_enabled   = true
+```
+If that is set to true, then you are using SSO, otherwise it's IAM.
+
+## Why should I care whether I am using IAM or SSO?
+Well, because even though both methods will try to get temporary AWS credentials, each method will use a different way to do that. In fact, Leverage relies on the AWS CLI to get the credentials and each method requires completely different commands to achieve that.
+
+## Do you have MFA enabled?
+MFA is optionally used via the IAM method. It can be enabled/disabled in the [build.env](https://github.com/binbashar/le-tf-infra-aws/blob/master/build.env) file.
+
+!!! info "Keep in mind that MFA should only be used with the IAM method, not with SSO."
+
+## Identify which credentials are failing
+Since Leverage actually relies on Terraform and, since most of the definitions are AWS resources, it is likely that you are having issues with the Terraform AWS provider, in other words, you might be struggling with AWS credentials. Now, bear in mind that Leverage can also be used with other providers such as Gitlab, Github, Hashicorp Cloud Platform, or even SSH via Ansiblel; so the point here is to understand what credentials are not working for you in order to focus the troubleshooting on the right suspect.
```

Comment:
```suggestion
Since Leverage actually relies on Terraform and, since most of the definitions are AWS resources, it is likely that you are having issues with the Terraform AWS provider, in other words, you might be struggling with AWS credentials. Now, bear in mind that Leverage can also be used with other providers such as Gitlab, Github, Hashicorp Cloud Platform, or even SSH via Ansible; so the point here is to understand what credentials are not working for you in order to focus the troubleshooting on the right suspect.
```